Dockerfile

```diff
@@ -1,32 +1,62 @@
 #
```

Comment:
Looks great at 1st blush.  A little unfortunate that there's so much repetition between the `Dockerfile` and `onbuild/Dockerfile`, but I'm not sure how to improve on that.  Need a bit more time to grok this, a lot going on here.